code/modules/minimap/minimap_renderer.dm

```diff
@@ -0,0 +1,157 @@
+		for (var/minimap_type in dynamic_area_directory)
+			if (!("[minimap_type]" in dynamic_area_overlays))
+				dynamic_area_overlays["[minimap_type]"] = list()
+			var/list/minimap_type_render_list = dynamic_area_overlays["[minimap_type]"]
+
+			var/list/area_groups = dynamic_area_directory[minimap_type]
+			for (var/area_group in area_groups)
+				var/icon/area_render = icon('icons/obj/minimap/minimap.dmi', "blank")
+				area_render.Scale(x_max, y_max)
+				area_render.SwapColor(rgb(0, 0, 0), rgb(0, 0, 0, 0))
+
+				var/list/area_list = area_groups[area_group]
+				for (var/turf/turf in area_list)
+					area_render.DrawBox(turf_color(turf), turf.x, turf.y)
+
+				var/atom/movable/area_render_object = new
+				area_render_object.icon = area_render
+				area_render_object.vis_flags = VIS_INHERIT_ID | VIS_INHERIT_LAYER
+				area_render_object.mouse_opacity = 0
+
+				minimap_type_render_list["[area_group]"] = area_render_object
+
+	///Separates a given bitflag into a list of the separate bits composing it. Eg. 1011 -> (0001, 0010, 1000), or in base 10: 11 -> (1, 2, 8)
+	proc/separate_bitflag(var/bitflag)
+		var/list/flag_list = list()
+		var/bit_length = ceil(log(2, bitflag + 1))
+
+		for (var/i = 0 to bit_length)
+			if (2**i & bitflag)
+				flag_list.Add(2**i)
+
+		return flag_list
+
+	///Checks whether a turf should be rendered on the minimap through the minimaps_to_render_on bitflag on /area.
+	proc/valid_turf(var/turf/T)
+		if (!T.loc)
+			return FALSE
+		var/area/A = T.loc
+		if (!A.minimaps_to_render_on)
+			return FALSE
+		return TRUE
+
+	///Determine the colour of a turf on the minimap through the station_map_colour variable on /turf.
+	proc/turf_color(turf/T)
+		if (!T.loc)
+			return
+		var/area/A = T.loc
+		return A.station_map_colour
+
+	///Generates an atom/movable for a specified minimap type.
+	proc/generate_minimap_render(var/minimap_type)
+		var/atom/movable/minimap_render_object = new()
+		var/icon/minimap_render = icon('icons/obj/minimap/minimap.dmi', "blank")
+		minimap_render.Scale(world.maxx, world.maxy)
+
+		var/flags = list()
+		flags += MAP_ALL
+		flags += src.separate_bitflag(minimap_type)
+
+		for (var/flag in flags)
+			var/list/area_list = src.dynamic_area_overlays["[flag]"]
+			for (var/area in area_list)
+				var/atom/movable/area_render = area_list[area]
+				minimap_render_object.vis_contents += area_render
+
+			if (!("[flag]" in src.minimap_type_renders))
+				continue
+
+			var/icon/minimap_type_render = src.minimap_type_renders["[flag]"]
+			minimap_render.Blend(minimap_type_render, ICON_OVERLAY)
+
+		minimap_render_object.icon = minimap_render
+		minimap_render_object.vis_flags = VIS_INHERIT_ID | VIS_INHERIT_LAYER
+		minimap_render_object.mouse_opacity = 0
+
+		return minimap_render_object
+
+	///Recolours a specified area group to a specified colour.
+	proc/recolour_area(var/area_group, var/colour)
```

Comment:
as much as it pains me as a fellow brit, we should probably stick to `color` for proc names at least (don't worry about comments tho)